### ISSUE TITLE
May disk's driver type need more checks

### ISSUE THREAD

**Issue by nevermore-muyi** [jaypipes/ghw#296]
As the code [https://github.com/jaypipes/ghw/blob/main/pkg/block/block_linux.go#L271](https://github.com/jaypipes/ghw/blob/main/pkg/block/block_linux.go#L271) shows, when I create a device mapper device which starts with **dm-** prefix, and the function [https://github.com/jaypipes/ghw/blob/main/pkg/block/block_linux.go#L316](https://github.com/jaypipes/ghw/blob/main/pkg/block/block_linux.go#L274) thinks it's a unknown device, but [(https://github.com/jaypipes/ghw/blob/main/pkg/block/block_linux.go#L274)](https://github.com/jaypipes/ghw/blob/main/pkg/block/block_linux.go#L274) may define it's a SSD.
![image](https://user-images.githubusercontent.com/12146934/150089782-b6a332ce-904e-4436-b400-9935eb53075b.png)

So, in my option, driver type may need more checks.

**Comment by jaypipes** [comment]
@nevermore-muyi hi! thanks for your submission!

Are you suggesting that instead of returning an UNKNOWN disk type, we should check for whether the disk is rotational, and if not, return SSD disk type?

**Comment by nevermore-muyi** [comment]
@jaypipes Maybe a little different. I suggest only KNOWN disk type needs more check if it's rotational and define it's a SSD, but UNKNOWN disk type, just return it?

**Comment by fromanirh** [comment]
I tend to agree the code should not set the drive type depending only on the value of `rotational`

**Comment by jaypipes** [comment]
@nevermore-muyi we are considering creating a separate package for devicemappings/devicemapper. [See conversation here](https://github.com/jaypipes/ghw/issues/298).

**Comment by nevermore-muyi** [comment]
> @nevermore-muyi we are considering creating a separate package for devicemappings/devicemapper. [See conversation here](https://github.com/jaypipes/ghw/issues/298).

Got it. Looks very cool.
